keps/sig-node/4781-kublet-restart-pod-status/README.md

```diff
@@ -0,0 +1,559 @@
+
+For the `Ready` state determination, the overall logic aligns with that of the Started state, with the distinction that we need to consider situations where the kubelet's restart time exceeds the node-monitor-grace-period or similar conditions. In these cases, the API server may set both the Node and Pod states to not ready, so we will not derive the Ready state from the container. Instead, we will retrieve the Ready status directly from pod.Status.Conditions.
+
+In summary, when a container is healthy and there are no exceptions, and the probe is present:
+- If the container has encountered an unexpected issue, for example, VM reboots or the container crashed, both the Start and Ready states will be determined based on the probe (as per existing logic)
+- If the kubelet's restart time is short, the Container Start and Ready states will both be True(or the previous states).
+- If the kubelet's restart time is long (exceeding `node-monitor-grace-period`), and the container is healthy, Start will be True, but Ready will be False.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[ ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+##### Unit tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `<package>`: `<date>` - `<test coverage>`
```

Comment:
You should add your test plan. I assume you've already implemented a POC, so this should be easy for you.

Reply:
This still applies.

Reply:
+1, this feature will definitely need strong test coverage and may not be straightforward to test. we should start planning that early.